packages/blade/src/components/SideNav/SideNavItems/SideNavLink.web.tsx

```diff
@@ -264,6 +265,8 @@ const SideNavLink = ({
                       isFirstRender: false,
                     });
                   }
+
+                  onClick?.(e);
```

Comment:
You might have to pass this in isL3Trigger block as well so that onClick is also supported on that collapsible sidenavlink of l3 trigger

Reply:
Yup, done.